sklearn/utils/validation.py

```diff
@@ -374,6 +378,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         if force_all_finite:
             _assert_all_finite(array)
 
+    if array.ndim == 1:
```

Comment:
Well, it was. It ensures that it was 2d if it didn't error ;)
We could rename it to ``error_if_not_2d`` but that seems pretty long, and during the deprecation it will warn and not error.

We could also redo it into ``allowed_ndim`` and merge it with ``allow_nd`` but that might be a bit tricky.